yt_dlp/extractor/tvopengr.py

```diff
@@ -0,0 +1,150 @@
+                    format_url, video_id, 'mp4', m3u8_id=format_id,
+                    fatal=False)
+            elif ext == 'mpd':
+                formats_, subs_ = self._extract_mpd_formats_and_subtitles(
+                    format_url, video_id, 'mp4', fatal=False)
+            else:
+                formats.append({
+                    'url': format_url,
+                    'format_id': format_id,
+                })
+                continue
+            formats.extend(formats_)
+            subs.update(subs_)
+        self._sort_formats(formats)
+        return formats, subs
+
+    def _real_extract(self, url):
+        scheme, netloc, video_id, display_id = self._match_valid_url(url).group('scheme', 'netloc', 'id', 'slug')
+        if netloc.find('tvopen.gr') == -1:
+            return self._return_canonical_url(url, video_id)
+        webpage = self._download_webpage(url, video_id)
+        info = self._search_json_ld(webpage, video_id, expected_type='VideoObject')
+        info['formats'], info['subtitles'] = self._extract_formats_and_subs(
+            self._download_api_data(video_id, scheme=scheme), video_id)
+        max_dimensions = max(
+            [tuple(format.get(k) or 0 for k in ('width', 'height')) for format in info['formats']],
+            default=None)
+        if max_dimensions[0]:
+            for thumbnail in info['thumbnails']:
+                thumbnail['url'] = re.sub(r'(/imgHandler/)\d+', rf'\g<1>{max_dimensions[0]}', thumbnail['url'])
+                thumbnail['width'], thumbnail['height'] = max_dimensions
+        description, _html = next(get_elements_text_and_html_by_attribute('class', 'description', webpage))
+        if description and _html.startswith('<span '):
+            info['description'] = description
+        info['id'] = video_id
+        info['display_id'] = display_id
+        return info
+
+
+class TVOpenGrEmbedIE(TVOpenGrBaseIE):
+    IE_NAME = 'tvopengr:embed'
+    IE_DESC = 'tvopen.gr embedded videos'
+    _VALID_URL = r'https?://(?:www\.|cdn\.|)(?:tvopen|ethnos).gr/embed/(?P<id>\d+)'
+
+    _TESTS = [{
+        'url': 'https://cdn.ethnos.gr/embed/100963',
+        'md5': '2da147881f45571d81662d94d086628b',
+        'info_dict': {
+            'id': '100963',
+            'display_id': 'koronoiosapotoysdieythyntestonsxoleionselftestgiaosoysdenbrhkan',
+            'title': 'md5:2c71876fadf0cda6043da0da5fca2936',
+            'description': 'md5:17482b4432e5ed30eccd93b05d6ea509',
+            'thumbnail': 'https://opentv-static.siliconweb.com/imgHandler/1920/5804e07f-799a-4247-a696-33842c94ca37.jpg',
+            'ext': 'mp4',
+            'upload_date': '20220108',
+            'timestamp': 1641600000,
+        },
+    }]
+
+    @classmethod
+    def _extract_urls(cls, webpage, origin_url=None):
+        # make the scheme in _VALID_URL optional
+        _URL_RE = r'(?:https?:)?//' + cls._VALID_URL.split('://', 1)[1]
```

Comment:
You can make it optional in _VALID_URL itself to simplify this. This is generally not done, but is also harmless